src/components/AbstractPageDefinitions.js

```diff
@@ -436,6 +442,70 @@ export default class AbstractPageDefinitions extends Component {
     )
   }
 
+  showVersionSelectorPopup(component, multiple) {
+    this.setState({ showVersionSelectorPopup: true, multipleVersionSelection: multiple, selectedComponent: component })
+  }
+
+  applySelectedVersions(versions) {
+    const { multipleVersionSelection, selectedComponent } = this.state
+    if (!multipleVersionSelection) {
+      return this.setState({ showVersionSelectorPopup: false }, async () => {
+        if (selectedComponent.changes) {
+          const key = `open${Date.now()}`
+          const NotificationButtons = (
+            <Fragment>
+              <AntdButton
+                type="primary"
+                size="small"
+                onClick={async () => {
+                  await this.onRemoveComponent(selectedComponent)
+                  await this.onAddComponent(
+                    EntitySpec.fromCoordinates({ ...selectedComponent, revision: versions, changes: {} })
```

Comment:
is it really `versions` (with an s)? how does that work?

Reply:
This function is built to accept either a single version (if the user used the Switch functionality) or an array (if the user used the Add functionality).
So this params could contain a single value or an array, it depends on the `multipleVersionSelection` into the component state

Reply:
ok but EntitySpecs are not intended/expected to have multiple entries in the revision property.

Reply:
Yes, it never happens.
In case of multiple versions selected, there's a loop that iterates all the versions, and then calls the specific function to retrieve it:

```
versions.map(version => {
        this.onAddComponent(EntitySpec.fromCoordinates({ ...selectedComponent, revision: version }))
      })
```